### PR TITLE
Upgrade operator-sdk to 0.15.1

### PR DESCRIPTION
**Pull request by sbose78**
* ( updated local CLI to 0.15.1 )
* Ran `operator-sdk print-deps` to understand what to pin with `replace(...)`
```
operator-sdk print-deps
module github.com/jenkinsci/kubernetes-operator

go 1.13

require (
        github.com/operator-framework/operator-sdk v0.15.1
        sigs.k8s.io/controller-runtime v0.4.0
)

// Pinned to kubernetes-1.16.2
replace (
        k8s.io/api => k8s.io/api v0.0.0-20191016110408-35e52d86657a
        k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.0.0-20191016113550-5357c4baaf65
        k8s.io/apimachinery => k8s.io/apimachinery v0.0.0-20191004115801-a2eda9f80ab8
        k8s.io/apiserver => k8s.io/apiserver v0.0.0-20191016112112-5190913f932d
        k8s.io/cli-runtime => k8s.io/cli-runtime v0.0.0-20191016114015-74ad18325ed5
        k8s.io/client-go => k8s.io/client-go v0.0.0-20191016111102-bec269661e48
        k8s.io/cloud-provider => k8s.io/cloud-provider v0.0.0-20191016115326-20453efc2458
        k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.0.0-20191016115129-c07a134afb42
        k8s.io/code-generator => k8s.io/code-generator v0.0.0-20191004115455-8e001e5d1894
        k8s.io/component-base => k8s.io/component-base v0.0.0-20191016111319-039242c015a9
        k8s.io/cri-api => k8s.io/cri-api v0.0.0-20190828162817-608eb1dad4ac
        k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.0.0-20191016115521-756ffa5af0bd
        k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.0.0-20191016112429-9587704a8ad4
        k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.0.0-20191016114939-2b2b218dc1df
        k8s.io/kube-proxy => k8s.io/kube-proxy v0.0.0-20191016114407-2e83b6f20229
        k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.0.0-20191016114748-65049c67a58b
        k8s.io/kubectl => k8s.io/kubectl v0.0.0-20191016120415-2ed914427d51
        k8s.io/kubelet => k8s.io/kubelet v0.0.0-20191016114556-7841ed97f1b2
        k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.0.0-20191016115753-cf0698c3a16b
        k8s.io/metrics => k8s.io/metrics v0.0.0-20191016113814-3b1a734dba6e
        k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.0.0-20191016112829-06bb3c9d77c9
)

replace github.com/docker/docker => github.com/moby/moby v0.7.3-0.20190826074503-38ab9da00309 // Required by Helm
replace github.com/openshift/api => github.com/openshift/api v0.0.0-20190924102528-32369d4db2ad // Required until https://github.com/operator-framework/operator-lifecycle-manager/pull/1241 is resolved
```

* `go mod tidy`, stumbled into  kubernetes/client-go#628, and therefore..
* Pinned `github.com/Azure/go-autorest => github.com/Azure/go-autorest v12.2.0+incompatible`  
* `make build` ( runs `operator-sdk generate k8s`, should really try to avoid generating code while building code, might give us unexpected results in CI :) )
* upgraded prometheus-operator dependency to v0.35.1 to fix 
```
# github.com/coreos/prometheus-operator/pkg/client/versioned/typed/monitoring/v1
../../go/pkg/mod/github.com/coreos/prometheus-operator@v0.29.0/pkg/client/versioned/typed/monitoring/v1/monitoring_client.go:87:32: undefined: serializer.DirectCodecFactory
```
* Code changes like below 

```
err = client.List(context.TODO(), roleBindings, k8sclient.InNamespace(namespace))
```

 to fix issues in reconciler[_test].go to fix 

```
pkg/controller/jenkins/configuration/base/reconcile.go:405:58: cannot use roleBindings (type *"k8s.io/api/rbac/v1".RoleBindingList) as type "sigs.k8s.io/controller-runtime/pkg/client".ListOption in argument to r.Configuration.Client.List:
        *"k8s.io/api/rbac/v1".RoleBindingList does not implement "sigs.k8s.io/controller-runtime/pkg/client".ListOption (missing ApplyToList method)
```

### CODE REVIEW

**Comment by waveywaves** [comment]
@sbose78 LGTM, the build is also passing successfully, after the updated version everything seems to be running fine. 
Wdyt @tomaszsek @jakalkhalili @pawelprazak

**Comment by jakalkhalili** [comment]
LGTM, but I think the @tomaszsek should look at this PR

**Comment by tomaszsek** [comment]
@sbose78 Please rebase with master because e2e tests didin't run. The fix for e2e test -  https://github.com/jenkinsci/kubernetes-operator/commit/018c8cc709ecc303a6e51b3b5699c6b047bfe318.

**Comment by sbose78** [comment]
done.

**Comment by tomaszsek** [comment]
@sbose78 Please look at Travis build.

**Comment by sbose78** [comment]
Pretty tight static checks I'd say ;)

File: pkg/controller/jenkins/configuration/user/seedjobs/seedjobs_test.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ import (
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/kubernetes/scheme"
 	"sigs.k8s.io/controller-runtime/pkg/client/fake"
-	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	logf "sigs.k8s.io/controller-runtime/pkg/log/zap"
```
Comment (sbose78) [inline pkg/controller/jenkins/configuration/user/seedjobs/seedjobs_test.go]: Change was needed because of an API deprecation notice.

**Comment by waveywaves** [comment]
@tomaszsek Can you check this out ?

**Pull request merged**